### PR TITLE
Run an Ethereum node

### PR DESCRIPTION
**Pull request by wackerow**
Introduces a beginner friendly page on running an Ethereum node:
https://ethereumorgwebsitedev01-runanode.gtsb.io/en/run-a-node/

Intended to be a resource for those who are less technically oriented and those who may have never considered running their own node before. Primary focus of this page is not on staking, though it can be used for users to get started with staking as well.

Reviews and feedback appreciated =)

- [x] Need to create an expandable info component for the "_Who_ should run a node?" section
- [x] Added links from `nodes-and-clients` and `nodes-and-clients/run-a-node` pages 
- [x] Update for "eth2" changes
- [x] Extract intl strings

Can also consider other places to link from

### CODE REVIEW

File: src/components/ExpandableCard.js
Comment (minimalsm) [inline src/components/ExpandableCard.js:67]: Not a fan of how we have to click directly on the 'More' text to expand the additional content - [example GIF](https://gyazo.com/574e7a8b9b8dd4b9a0cf7712908574cc). This is quite a small target and [could be difficult for mobile users to interact with](https://www.nngroup.com/articles/touch-target-size/).

Comment (minimalsm) [reply src/components/ExpandableCard.js:67]: I think it'd be an improvement if we make the entire Card clickable - [example GIF](https://gyazo.com/8f4f0e9f028d08572dd5048df9db3bb4). 

I've checked and we also use this on the History page and the [staking community grants page](https://ethereum.org/en/community/grants/staking-community-grants), neither are negatively affected by the change.

File: src/components/ExpandableCard.js
Comment (minimalsm) [inline src/components/ExpandableCard.js:106]:
```suggestion
    <Card onClick={() => setIsVisible(!isVisible)}>
```

File: src/components/ExpandableCard.js
Comment (minimalsm) [inline src/components/ExpandableCard.js:115]:
```suggestion
        <ButtonContainer>
```

Comment (minimalsm) [reply src/components/ExpandableCard.js:115]: If we agree on this change we also need to add `pointer: cursor` to the `Card` styled-component within ExpandableCard.js

File: src/pages/run-a-node.js
Comment (minimalsm) [inline src/pages/run-a-node.js:424]: I don't think we should have two primary action buttons here. [More info on best practices.](https://balsamiq.com/learn/articles/button-design-best-practices/#2-visually-differentiate-primary-and-secondary-actions)

```suggestion
        pathId: "getting-started",
        isSecondary: "isSecondary",
```

File: src/pages/run-a-node.js
Comment (minimalsm) [inline src/pages/run-a-node.js:359]:
```suggestion
  text-align: center;
  cursor: pointer;
```

File: src/pages/run-a-node.js
Comment (minimalsm) [inline src/pages/run-a-node.js:320]: This button doesn't pass accessibility contrast tests. Noticed you're using the Dapp node colours for this which I think is a nice touch—maybe we could tweak it slightly to make it compliant? 

Screenshot 2022-01-07 at 23 15 15

File: src/intl/en/page-run-a-node.json
Comment (minimalsm) [inline src/intl/en/page-run-a-node.json:110]: Feel like dropping the brackets here is a bit cleaner.

Screenshot 2022-01-07 at 23 27 38

Screenshot 2022-01-07 at 23 28 00


```suggestion
  "page-run-a-node-who-preview": "Nodes are not just for miners and validators. <i>Anyone</i> can run a node—you don't even need ETH.",
```

File: src/intl/en/page-run-a-node.json
Comment (minimalsm) [inline src/intl/en/page-run-a-node.json:47]:
```suggestion
  "page-run-a-node-getting-started-hardware-1": "You'll need some hardware to get started. Although running node software is possible on a personal computer, having a dedicated machine can greatly enhance the performance of your node while minimizing its impact on your primary computer.",
```

File: src/intl/en/page-run-a-node.json
Comment (minimalsm) [inline src/intl/en/page-run-a-node.json:48]:
```suggestion
  "page-run-a-node-getting-started-hardware-2": "When selecting hardware, consider that the chain is continually growing, and maintenance will inevitably be needed. Increasing specs can help delay the need for node maintenance.",
```

File: src/pages/run-a-node.js
Comment (minimalsm) [inline src/pages/run-a-node.js:623]: This feels too much text to be a 'small',

Comment (minimalsm) [reply src/pages/run-a-node.js:623]: > Though you can still technically get away with 1 TB, the blockchain continues to grow over time and this will soon no longer be enough.

Also, with the statelessness and state-expiry roadmap, I'm not sure if we should mention this. The idea of having to continually upgrade your node hardware indefinitely might put some potential node runners off.

File: src/pages/run-a-node.js
Comment (minimalsm) [inline src/pages/run-a-node.js:756]: This section feels a bit cramped to me. One thought I had was maybe making it a bit narrower similar to the '[How to stake](https://ethereum.org/en/eth2/staking)' section on our staking page?

**Proposed**
Screenshot 2022-01-07 at 23 42 07

**Current**
Screenshot 2022-01-07 at 23 42 15

Comment (minimalsm) [reply src/pages/run-a-node.js:756]: Don't love this solution but 🤷‍♂️

**Review by minimalsm — CHANGES_REQUESTED** [review]
Hey @wackerow this is looking awesome.

Left a few comments largely around small styling tweaks. Made a couple of suggestions on the content but will look more over weekend—takes forever to review the Translation JSONs 😬

File: src/components/ExpandableCard.js
Comment (wackerow) [reply src/components/ExpandableCard.js:67]: Yeah, personally prefer being able to click the entire card as well. Tried to start by reusing components, but would be down to modify this.

File: src/intl/en/page-run-a-node.json
Comment (wackerow) [reply src/intl/en/page-run-a-node.json:110]: Agreed, the ol' em dash alternative

File: src/pages/run-a-node.js
Comment (wackerow) [reply src/pages/run-a-node.js:624]: Remove it and just leave "2 TB" as the recommended minimum?

I'd not against that, but if you're suggesting changing it to just 1 TB I'd be hesitant until statelessness/expiry are actually live. 

This is a tricky section, since each client is a little different, and future-proofing / trying to minimize future maintenance is somewhat subjective. In the end I'd prefer to keep it simple, so I'm down to just keep it at "2 TB SSD" and leave the note that SSD speeds are essential.

File: src/pages/run-a-node.js
Comment (wackerow) [reply src/pages/run-a-node.js:757]: Yeah wouldn't be against adding some horizontal margin on those divs when on suitable screen sizes.

File: src/components/ExpandableCard.js
Comment (wackerow) [reply src/components/ExpandableCard.js:115]: Only hesitation with doing this... It's nice for opening/closing, but if you want to select any of the expanded text (at least on desktop with a mouse), it will close the card when you un-click to finish your selection.

I tried it by moving `onClick={() => setIsVisible(!isVisible)}` to `<Content>` instead, which works pretty well by only closing when you click the top section of the card, but it does leave the padding un-clickable. Curious your thought on this.

File: src/intl/en/page-run-a-node.json
Comment (taxmeifyoucan) [inline src/intl/en/page-run-a-node.json:103]: > downloads a copy of the Ethereum blockchain, keeps it up-to-date with new blocks and transactions '

One important fact I would add here is that client (cryptographically, mathematically) verifies this data.

File: src/pages/run-a-node.js
Comment (minimalsm) [reply src/pages/run-a-node.js]: I think we should say 2 TB and hope state bloat is fixed before they'll need to upgrade 😅

Comment (taxmeifyoucan) [reply src/pages/run-a-node.js]: You can also refer table under this https://ethereum.org/en/developers/docs/nodes-and-clients/#recommended-specifications
I would keep just 1TB as minimum, with prunning of old states, dropping ancient data, 1TB will be more than enough.

File: src/intl/en/page-run-a-node.json
Comment (wackerow) [reply src/intl/en/page-run-a-node.json:103]: Thanks for the feedback @taxmeifyoucan .. Yeah, can't disagree too strongly. Been a little back and forth on this statement, trying to keep it really simple for people in this section. Will keep this in mind for sure as we get other feedback. 👍🏻

File: src/components/ExpandableCard.js
Comment (wackerow) [reply src/components/ExpandableCard.js:115]: Tried this out... not sure I love it enough to keep it in for this PR, so reverted it back. The "ExpandableCard" is an existing component, which if we'd like to revamp it I have no problem with that, but requesting we address that separately to limit scope creep here.

File: src/pages/run-a-node.js
Comment (wackerow) [reply src/pages/run-a-node.js]: I think I'm in favor of just recommending 2TB here. At the risk of excluding some people who are willing/able to get a 1 TB but not a 2 TB drive, recommending 2 will minimize future issues and complications in the attempt to keep this geared as much as possible towards non-technical users. This will also keep things really simple as we approach the merge, where the Beacon Chain will also need to be synced. 

In the future as statelessness, and history/state expiry become a reality, we can certainly readdress this minimum, and hopefully drop it entirely someday. In the meantime, of course more complex setups exist to bring cost down, but again this guide is intended to keep things as simple as possible, and I think this is a reasonable way to help accomplish this.

File: src/pages/run-a-node.js
Comment (samajammin) [inline src/pages/run-a-node.js:590]: Rather than:
- "Getting started - Part 1: Hardware"
- "Part 2: Software"

 I think I'd prefer a structure of:
- "Get started"
  - "Part 1: Hardware"
  - "Part 2: Software"

File: src/intl/en/page-run-a-node.json
Comment (samajammin) [reply src/intl/en/page-run-a-node.json:103]: I agree - came here to say that. Maybe something like:

> Known as a "client," this software downloads a copy of the Ethereum blockchain, keeps it up to date by verifying new activity, and helps others download and update their own secure copies.

Comment (samajammin) [reply src/intl/en/page-run-a-node.json:103]: On the other hand, I'm pretty fine with what you have. I'm guessing these users will understand what transactions & blocks are...

Comment (wackerow) [reply src/intl/en/page-run-a-node.json:103]: Just tossing out another take

> Known as a \"client,\" this software downloads a copy of the Ethereum blockchain and verifies the validity of every block, then keeps it up-to-date with new blocks and transactions, and helps others download and update their own copies.
```suggestion
  "page-run-a-node-what-1-text": "Known as a \"client,\" this software downloads a copy of the Ethereum blockchain and verifies the validity of every block, then keeps it up-to-date with new blocks and transactions, and helps others download and update their own copies.",
```

File: src/components/ExpandableCard.js
Comment (samajammin) [reply src/components/ExpandableCard.js:84]: I tend to agree but think we can keep this as a separate PR, given that we use this card component all over the place.

File: src/components/ExpandableCard.js
Comment (samajammin) [reply src/components/ExpandableCard.js:132]: Agreed. As I noted above, this is why I think we should address that separately.

File: src/pages/run-a-node.js
Comment (samajammin) [reply src/pages/run-a-node.js]: Agree with @wackerow's last comment. I'd rather err on the side of caution.

File: src/intl/en/page-run-a-node.json
Comment (samajammin) [reply src/intl/en/page-run-a-node.json:103]: LGTM @wackerow 👍

File: src/pages/run-a-node.js
Comment (samajammin) [reply src/pages/run-a-node.js:590]: @wackerow noting this still needs to be addressed.

**Comment by samajammin** [comment]
Image 2022-01-19 at 12 00 19 PM

While I like the structure of this section I wonder if these sentence fragments ("With hardware." & "While online.") will create potential translations issues. Curious what @lukassim thinks.

File: src/intl/en/page-run-a-node.json
Comment (samajammin) [inline src/intl/en/page-run-a-node.json:105]:
```suggestion
  "page-run-a-node-what-2-text": "Ethereum is designed to be able to run a client on average consumer-grade computers. You can use any personal computer, but most users opt to run their node on dedicated hardware to eliminate the performance impact on their machine and minimize node downtime.",
```

File: src/intl/en/page-run-a-node.json
Comment (samajammin) [inline src/intl/en/page-run-a-node.json:107]: Should we mention the why here? e.g. "merely the act of continuously running client software on a computer while connected to the internet, so that your computer can communicate and sync with other peers in the network."

File: src/intl/en/page-run-a-node.json
Comment (samajammin) [inline src/intl/en/page-run-a-node.json:109]:
```suggestion
  "page-run-a-node-who-preview": "Everyone! Nodes are not just for miners and validators. <i>Anyone</i> can run a node—you don't even need ETH.",
```

File: src/intl/en/page-run-a-node.json
Comment (samajammin) [inline src/intl/en/page-run-a-node.json:110]:
```suggestion
  "page-run-a-node-who-copy-1": "You don't need to stake ETH or be a miner to run a node. In fact, it's every other node on Ethereum that holds miners and validators accountable.",
```

File: src/intl/en/page-run-a-node.json
Comment (samajammin) [inline src/intl/en/page-run-a-node.json:111]:
```suggestion
  "page-run-a-node-who-copy-2": "You may not get the financial rewards that validators and miners earn, but there are many other benefits of running a node - both for you and the network - that every Ethereum user should consider, including privacy, security, reduced reliance on third-party servers, censorship resistance and improved health and decentralization of the network.",
```

**Comment by samajammin** [comment]
Dropped in a few comments but didn't finish. Will come back to this tonight! Feel free to hold off on making changes until I've finished.

File: src/components/ExpandableCard.js
Comment (samajammin) [reply src/components/ExpandableCard.js:84]: @wackerow please create a separate story in this epic to address this.

File: src/components/FeedbackCard.js
Comment (samajammin) [inline src/components/FeedbackCard.js:44]: We should translate this string.

**Comment by samajammin** [comment]
I don't follow this section in red:
Image 2022-01-25 at 7 43 08 PM
I think we could just remove it entirely? We could update the 2nd bullet to "App-like setup with a GUI."

**Review by samajammin — CHANGES_REQUESTED** [review]
So close! Made some minor copy changes. A few comments to address. Once those are made, I think we're ready to ship 🚢

**Comment by wackerow** [comment]
@samajammin So the above was accidentally incomplete copy... Should've said "small in size, but come fully loaded."
Somewhat of an after thought, but was hoping to include this as another subtle reminder that we're not talking large mining rigs... many NUCs only have about an 11 sq cm footprint, standing <5 cm tall.
Thoughts on if we trim out the additional comment about choosing your hardware, move the GUI comment, and leave this small callout?
image

**Comment by wackerow** [comment]
Also, still have two checkboxes before merging... intl strings and eth2 updates. Will get these going in a branch.

**Comment by samajammin** [comment]
@wackerow my take - I we just remove that "these solutions are small in size, but come fully loaded." section & image. We already mention "fully loaded" in the same card above. We also mention nodes run on consumer hardware earlier in the page. I feel like mentioning this in the "fully loaded" solution implies that's not the case for the "build your own".

**Comment by samajammin** [comment]
@wackerow also FWIW I'd say we could ship this page without full intl support. I think it'd be good to collect some community feedback (& potentially make tweaks) before we upload the page to Crowdin for translation.

**Comment by wackerow** [comment]
@samajammin Okay cool, https://github.com/ethereum/ethereum-org-website/pull/5226 has the latest here. After we get that pulled into this branch, should be ready for final review before prod.

**Review by samajammin — APPROVED** [review]
🚀

**Comment by Drblessing** [comment]
This resource was a great help setting up a node. Could there be more resources about exposing a client node's RPC/HTTP server for a small remote community? Right now in [Spin up your own Ethereum node](https://ethereumorgwebsitedev01-runanode.gtsb.io/en/developers/docs/nodes-and-clients/run-a-node/#spinning-up-node), it mentions port forwarding the router and setting `--http.addr 0.0.0.0`. It also mentions routing it through tor, or a web server like Nginx, and a separate guide for this would be much appreciated.

**Comment by jjmstark** [comment]
@wackerow I think we should probably change this page a bit to make clear that dappnode is just one company that offers a solution. 

In particular I am thinking of this section:
Screen Shot 2022-04-14 at 12 07 32 PM

This feels like an "endorsement" compared to how we usually show projects/companies on a page. We also use the Dappnode logo in this section in a way that does not make it clear to the reader that we are using someone's logo, and not just an image representing "running a node".  It is only lower on the page that we show another option (Avado). 

Maybe a way to change it would be to instead have this section communicate something like "Now we have simple node-running services like Dappnode or Avado that gives users an app-like experience while managing their node."

Same point about the "find some helpers" section, maybe we ought to be linking to more than just the dappnode community?

**Pull request merged**
